### ISSUE TITLE
Network Load Balancer support Security Groups

### ISSUE THREAD

**Issue by Saff-Buraq-Dev** [aws-cloudformation/cfn-lint#2868]
### CloudFormation Lint Version

0.79.7

### What operating system are you using?

Ubuntu

### Describe the bug

[Network Load Balancer now supports security groups](https://aws.amazon.com/about-aws/whats-new/2023/08/network-load-balancer-supports-security-groups/)
This rule needs to be updated: E2503 Security groups are not supported for load balancers with type "network"

### Expected behavior

Attaching a Security Group to an NLB should not raise this warning.

### Reproduction template

```yaml
AWSTemplateFormatVersion: 2010-09-09

Parameters:
  paramLBSecurityGroup:
    Type: String
    Default: auto
    AllowedPattern: ^sg-[0-9a-f]+$

  paramLBSubnets:
    Type: List<AWS::EC2::Subnet::Id>

Resources:
  LoadBalancer:
      Type: 'AWS::ElasticLoadBalancingV2::LoadBalancer'
      Properties:
        Scheme: internal
        SecurityGroups: !Ref paramLBSecurityGroup
        Subnets: !Ref paramLBSubnets
        Type: network
```

**Comment by kddejong** [comment]
Already resolved.  Update to https://github.com/aws-cloudformation/cfn-lint/releases/tag/v0.79.8
